projects/core/src/routing/facade/routing.service.ts

```diff
@@ -79,6 +81,27 @@ export class RoutingService {
     return this.navigate(path, query, extras);
   }
 
+  /**
+   * Resolves the absolute url for the given `UrlCommands`.
+   *
+   * The absolute URL uses the origin of the current location.
+   */
+  getUrl(commands: UrlCommands): string {
+    if (!this.router) {
+      throwError('no router available to create the url tree');
+      return;
+    }
+    let url = this.winRef.document.location.origin;
```

Comment:
Could you double check if `winRef.document.location.origin` will return you a correct value in SSR?
I believe it was not possible some time ago, so I introduced `SERVER_REQUEST_ORIGIN` (and `SERVER_REQUEST_URL` injection token).

--

Anyway, to build canonical URL feature in SSR may might not need to write this whole method, as you can access  the full request URL via `SERVER_REQUEST_URL` injection token (works only in SSR).

Reply:
I've resolved these values in the WindowRef class, happy to get your opinion on this.